analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCallGCSuppressFinalize.NetCore.cs

```diff
@@ -15,13 +15,61 @@ public async ValueTask DisposeAsync()
         {
             await DisposeAsyncCore();
 
-            GC.SuppressFinalize(this); // Noncompliant - FP, see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method
+            GC.SuppressFinalize(this); // Compliant, see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method
         }
 
         protected virtual ValueTask DisposeAsyncCore() => default;
     }
 
-    public class Fake
+    public class Explicit : IAsyncDisposable
+    {
+        public void Method()
+        {
+            GC.SuppressFinalize(this); // Noncompliant
+        }
+
+        async ValueTask IAsyncDisposable.DisposeAsync()
+        {
+            await DisposeAsyncCore();
+
+            GC.SuppressFinalize(this); // Compliant, see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#the-disposeasync-method
+        }
+
+        protected virtual ValueTask DisposeAsyncCore() => default;
+    }
+
+    public class FakeA
+    {
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore();
```

Comment:
do we have a rule for `await DisposeAsyncCore();` ? to make sure `DisposeAsync` is properly implemented?

Reply:
I opened https://github.com/SonarSource/sonar-dotnet/issues/3853